packages/react-native-passkey-stamper/src/index.ts

```diff
@@ -0,0 +1,207 @@
+      authenticatorSelection: config.authenticatorSelection || {
+        requireResidentKey: true,
+        residentKey: "required",
+        userVerification: "preferred",
+      },
+      attestation: config.attestation || "none",
+      extensions: config.extensions || {},
+      // All algorithms can be found here: https://www.iana.org/assignments/cose/cose.xhtml#algorithms
+      // We only support ES256 and RS256, which are listed below
+      pubKeyCredParams: [
+        {
+          type: "public-key",
+          alg: -7,
+        },
+        {
+          type: "public-key",
+          alg: -257,
+        },
+      ],
+    },
+    options
+  );
+
+  return {
+    authenticatorName: config.authenticatorName,
+    challenge: challenge,
+    attestation: {
+      credentialId: base64Tobase64url(registrationResult.id),
+      clientDataJson: base64Tobase64url(
+        registrationResult.response.clientDataJSON
+      ),
+      attestationObject: base64Tobase64url(
+        registrationResult.response.attestationObject
+      ),
+      // TODO: can we infer the transport from the registration result?
+      // In all honesty this isn't critical so we default to "hybrid" because that's the transport used by passkeys.
+      transports: ["AUTHENTICATOR_TRANSPORT_HYBRID"],
+    },
+  };
+}
+
+/**
+ * Stamper to use with `@turnkey/http`'s `TurnkeyClient`
+ */
+export class PasskeyStamper {
+  rpId: string;
+  timeout: number;
+  userVerification: UserVerificationRequirement;
+  allowCredentials: PublicKeyCredentialDescriptor[];
+  extensions: Record<string, unknown>;
+
+  constructor(config: TPasskeyStamperConfig) {
+    this.rpId = config.rpId;
+    this.timeout = config.timeout || defaultTimeout;
+    this.userVerification = config.userVerification || defaultUserVerification;
+    this.allowCredentials = config.allowCredentials || [];
+    this.extensions = config.extensions || {};
+  }
+
+  async stamp(payload: string) {
+    const challenge = getChallengeFromPayload(payload);
+
+    const signingOptions = {
+      challenge: challenge,
+      rpId: this.rpId,
+      timeout: this.timeout,
+      allowCredentials: this.allowCredentials,
+      userVerification: this.userVerification,
+      extensions: this.extensions,
+    };
+
+    const authenticationResult = await Passkey.authenticate(signingOptions);
+
+    const stamp = {
+      authenticatorData: base64Tobase64url(
+        authenticationResult.response.authenticatorData
+      ),
+      clientDataJson: base64Tobase64url(
+        authenticationResult.response.clientDataJSON
+      ),
+      credentialId: base64Tobase64url(authenticationResult.id),
+      signature: base64Tobase64url(authenticationResult.response.signature),
+    };
+
+    return {
+      stampHeaderName: stampHeaderName,
+      stampHeaderValue: JSON.stringify(stamp),
+    };
+  }
+}
+
+/**
+ * Simple util to convert a base64-encoded string to base64url
+ */
+function base64Tobase64url(s: string): string {
```

Comment:
fine with leaving it since it's only used here, but i suppose this is candidate for moving into `src/util.ts`?

Reply:
Oh right! That's a good call. I'll move it.